tests/test_core.py

```diff
@@ -4948,7 +4948,7 @@ def test_unistd_truncate(self):
       self.do_run(src, expected, js_engines=[NODE_JS])
 
   @no_windows("Windows throws EPERM rather than EACCES or EINVAL")
-  @unittest.skipIf(os.geteuid() == 0, "Root access invalidates this test by being able to write on readonly files")
+  @unittest.skipIf(WINDOWS or os.geteuid() == 0, "Root access invalidates this test by being able to write on readonly files")
```

Comment:
this change looks unrelated? but i guess this was an issue you saw on windows?

Reply:
Oh yeah, sorry, forgot to make a note about this: The previous form caused the test suite to fail on Windows altogether, the line 

```python
@unittest.skipIf(os.geteuid() == 0, "Root access invalidates this test by being able to write on readonly files")
```

fails on Windows with an error "geteuid() is not a function" when python is parsing the file in loading time, i.e. already before executing any test code. The change makes the code avoid evaluating `os.geteuid()` on Windows altogether.